podium-gateway/src/main/java/nl/thehyve/podium/service/RequestService.java

```diff
@@ -640,13 +671,34 @@ public RequestRepresentation requestRevision(
         // TODO: Aggregate mails for multiple organisations per user.
 
         for (UUID organisationUuid: request.getOrganisations()) {
-            // TODO: validate request type of the request with the supported request types of the organisation.
             Request organisationRequest = requestMapper.clone(request);
 
             // Create organisation revision details
             RequestDetail revisionDetail = requestDetailMapper.clone(request.getRequestDetail());
             organisationRequest.setRevisionDetail(revisionDetail);
 
+            try {
+                Set<RequestType> selectedRequestTypes = organisationRequest.getRequestDetail().getRequestType();
+
+                // Fetch the organisation object and filter the organisation supported request types.
+                OrganisationDTO organisationDTO = organisationClientService.findOrganisationByUuid(organisationUuid);
+
+                Set<RequestType> organisationSupportedRequestTypes
```

Comment:
This looks like a complicated way of doing set intersection. If that is the intention, please move it to a separate (static) function. Guava has a function for this: [Sets.intersection](https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/Sets.html#intersection-java.util.Set-java.util.Set-).